### PR TITLE
Convert website scan result provider to support high concurrency write storage operation

### LINKED ISSUE #1306: [Bug] Resubmitted scan shows status stuck at In progress
## Describe the bug

Executed new scan to avoid the NotFound response issue in #1295 but the scan ids stuck with status In progress.

## Area

- [ ] Install script
- [ ] Update script
- [x] Service

## Expected behavior

Scan completes

### PR DESCRIPTION
**Pull request by lamaks**
#### Details

Converted the `WebsiteScanResult` document to be stored in multiple DB documents. The common data is stored in a single instance of the common `WebsiteScanResultBase` DB document with the composite key (`baseUrl`, `scanGroupId`). The frequent delta updates are stored in multiple `WebsiteScanResultPart` DB documents with the composite key (`baseId`, `scanId`). Each `WebsiteScanResultPart` DB document corresponds to the scan that created the update. The `WebsiteScanResult` document corresponding read operation will merge delta updates in a memory for consuming by application layer.

##### Motivation

Fix scan requests Cosmos DB trigger function processing error due to high concurrency write storage operation to the single `WebsiteScanResult` document instance

#### Pull request checklist

- [x] Addresses an existing issue: Fixes #1306, #1295
- [x] Added relevant unit test for your changes. (`yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] Ran precheckin (`yarn precheckin`)
- [x] Validated in an Azure resource group

### CODE REVIEW

File: packages/service-library/src/data-providers/website-scan-result-provider.ts [hunk 1]
```diff
@@ -54,29 +69,43 @@ export class WebsiteScanResultProvider {
...
-                } else {
-                    return this.mergeAndWrite(operationResult.scanResult, websiteScanResult);
-                }
-            },
+            async () => this.mergeOrCreateImpl(dbDocument),
             async (err) =>
                 this.logger.logError(`Failed to update website scan result Cosmos DB document. Retrying on error.`, {
-                    document: JSON.stringify(scanResultNormalized),
+                    baseId: dbDocument.baseDocument.id,
+                    partId: dbDocument?.partDocument.id,
```
Comment (pownkel) [inline packages/service-library/src/data-providers/website-scan-result-provider.ts:80]: Should the first line here also have a null check for dbDocument?

Comment (lamaks) [reply packages/service-library/src/data-providers/website-scan-result-provider.ts]: updated

**Review by AhmedAbdoOrtiga — CHANGES_REQUESTED** [review]
LGTM, can we submit many requests under same report group id? I just wanna simulate the case when the batch feed trigger fails because of concurrent t write to the same webRef document and make sure it's fixed

**Comment by lamaks** [comment]
> LGTM, can we submit many requests under same report group id? I just wanna simulate the case when the batch feed trigger fails because of concurrent t write to the same webRef document and make sure it's fixed

Yes, this is what was tested and targeted for. The refactoring allow to skip writes for the same content db document and have frequent changes being written into separate db documents to avoid db document merge on write hence avoid write concurrency issue.

**Review by AhmedAbdoOrtiga — APPROVED** [review]

**Pull request merged**